### PR TITLE
Fixes #39901 for RH/CentOS 7

### LINKED ISSUE #39901: network.managed ipaddrs ignored
### Description of Issue/Question
Using network.managed for any interface and using ippaddrs, all ips w/in the ipaddrs list are ignored instead of generating a IPADDR0...X and GATEWAY/SUBNET0...X

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)
```
lo:
  network.managed:
    - name: lo
    - type: eth
    - onboot: yes
    - userctl: no
    - ipaddrs:
      - 127.0.0.1/8
      - 10.234.0.73/32
```
### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)
Use above state and note that your original ifcfg-lo which would look like this:
```
DEVICE=lo
IPADDR=127.0.0.1
NETMASK=255.0.0.0
NETWORK=127.0.0.0
# If you're having problems with gated making 127.0.0.0/8 a martian,
# you can change this to something else (255.255.255.255, for example)
BROADCAST=127.255.255.255
ONBOOT=yes
NAME=loopback
```

now looks like this:
```
DEVICE="lo"
HWADDR="00:00:00:00:00:00"
USERCTL="no"
TYPE="Ethernet"
ONBOOT="yes"
```

instead of something more realistic like this:
```
DEVICE="lo"
HWADDR="00:00:00:00:00:00"
USERCTL="no"
TYPE="Ethernet"
ONBOOT="yes"
IPADDR0="127.0.0.1"
NETMASK0="255.0.0.0"
IPADDR1="10.234.0.73"
NETMASK1="255.255.255.255"
```

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
Salt Version:
           Salt: 2016.11.3
 
Dependency Versions:
           cffi: 0.8.6
       cherrypy: 3.2.2
       dateutil: 1.5
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: 1.4.3
       M2Crypto: 0.21.1
           Mako: 0.8.1
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: 1.2.5
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Sep 15 2016, 22:37:39)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 4.9.13-2.el7.centos.x86_64
         system: Linux
        version: CentOS Linux 7.2.1511 Core
```
Even on salt v2016.3.5, I get the same results, and it doesn't matter what IP ranges, subnets, interfaces, etc.. I choose.  Also, I used the example straight from https://docs.saltstack.com/en/latest/ref/states/all/salt.states.network.html#configuration-of-network-interfaces for configuration the loopback device and same results.  I'm trying to avoid keep doing the rc.local hack to add VIPs to servers or having to setup an ifcfg-lo template when ideally ipaddrs for network.managed should work.

### PR DESCRIPTION
**Pull request by seanjnkns**
### What does this PR do?
Corrects the functionality of ipaddrs and ipv6addrs in network.managed state for RH/CentOS 7.

### What issues does this PR fix or reference?
#39901

### Previous Behavior
ipaddrs and ipv6addrs opts were passed to rh_ip.py module, but were ignored in the template.

### New Behavior
ipaddrs and ipv6addrs now correctly populate the template

### Tests written?

No

### Commits signed with GPG?

Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by cachedout** [comment]
@seanjnkns I would love to start to get this functionality under test. Any chance you would be willing to add some for this functionality?

File: salt/templates/rh_ip/rh7_eth.jinja [hunk 1]
```diff
@@ -15,16 +15,19 @@ DEVICE="{{name}}"
 {%endif%}{% if onparent %}ONPARENT={{onparent}}
 {%endif%}{% if ipv4_failure_fatal %}IPV4_FAILURE_FATAL="{{ipv4_failure_fatal}}"
 {%endif%}{% if ipaddr %}IPADDR="{{ipaddr}}"
-{%endif%}{% if ipaddr_start %}IPADDR_START="{{ipaddr_start}}"
-{%endif%}{% if ipaddr_end %}IPADDR_END="{{ipaddr_end}}"
```
Comment (gtmanfred) [inline salt/templates/rh_ip/rh7_eth.jinja]: This should not be removed, these are valid values for CentOS/RHEL 7.

```
[root@salt ~]# grep IPADDR_START /etc/sysconfig/network-scripts/ifup-aliases
#    IPADDR_START    -- ipaddr to start range at. eg "192.168.30.1"
# Note that IPADDR_START and IPADR_END are required to be in the same class-c
# block. I.e. IPADDR_START=192.168.30.1 and IPADDR_END=192.168.31.255 is
    ipaddr_prefix=${IPADDR_START%.*}
    ipaddr_startnum=${IPADDR_START##*.}
    if [ "${IPADDR_START%.*}" != "${IPADDR_END%.*}" ]; then
        net_log $"error in $FILE: IPADDR_START and IPADDR_END don't agree"
        net_log $"error in $FILE: IPADDR_START greater than IPADDR_END"
```

**Review by gtmanfred — CHANGES_REQUESTED** [review]

**Comment by seanjnkns** [comment]
@gtmanfred: These are already removed.  The git commit is based off 2016.11.  If I did it off of 2016.11.8, as I would have preferred but your contributor docs ask to go off of the top branch for a release, you wouldn't have seen that.  If it helps, I can resubmit the PR against 2016.11.8.

**Comment by seanjnkns** [comment]
@gtmanfred, alright, found your commit of a7d3d668f4a5eca52f84dd03772ed0ca8e545c47 where it was added.  Looks like this was submitted on Oct 19th, which was after I started putting everything together.  I'll merge in your changes.

**Comment by seanjnkns** [comment]
I'll resubmit a PR and @cachedout, I'll see about possibly adding tests.  My concern is that the current rh_ip_test.py is lacking numerous tests in my opinion, and I've always had a hard time following the logic of the test suite, let alone writing for them, given the lack of any formal documentation on how to utilize it.  If you could point me in the direction of where there is some good documentation to understand better how salt utilizes the MagicMock test suite, it'd make the process a lot smoother.  For reference, I've read https://docs.saltstack.com/en/latest/topics/development/tests/index.html numerous times, and always felt it to be incomplete.

**Pull request closed without merging**